derive_client/clients/base_client.py

```diff
@@ -833,3 +838,258 @@ def transfer_from_subaccount_to_funding(self, amount: int, asset_name: str, suba
+        maker_nonce = base_nonce
+        taker_nonce = base_nonce + 1
+
+        # Create maker action (sender)
+        maker_action = SignedAction(
+            subaccount_id=from_subaccount_id,
+            owner=self.wallet,
+            signer=self.signer.address,
+            signature_expiry_sec=MAX_INT_32,
+            nonce=maker_nonce,
+            module_address=self.config.contracts.TRADE_MODULE,
+            module_data=MakerTransferPositionModuleData(
+                asset_address=instrument["base_asset_address"],
+                sub_id=int(instrument["base_asset_sub_id"]),
+                limit_price=transfer_price,
+                amount=transfer_amount,
+                recipient_id=from_subaccount_id,
+                position_amount=original_position_amount,
+            ),
+            DOMAIN_SEPARATOR=self.config.DOMAIN_SEPARATOR,
+            ACTION_TYPEHASH=self.config.ACTION_TYPEHASH,
+        )
+
+        # Create taker action (recipient)
+        taker_action = SignedAction(
+            subaccount_id=to_subaccount_id,
+            owner=self.wallet,
+            signer=self.signer.address,
+            signature_expiry_sec=MAX_INT_32,
+            nonce=taker_nonce,
+            module_address=self.config.contracts.TRADE_MODULE,
+            module_data=TakerTransferPositionModuleData(
+                asset_address=instrument["base_asset_address"],
+                sub_id=int(instrument["base_asset_sub_id"]),
+                limit_price=transfer_price,
+                amount=transfer_amount,
+                recipient_id=to_subaccount_id,
+                position_amount=original_position_amount,
+            ),
+            DOMAIN_SEPARATOR=self.config.DOMAIN_SEPARATOR,
+            ACTION_TYPEHASH=self.config.ACTION_TYPEHASH,
+        )
+
+        # Sign both actions
+        maker_action.sign(self.signer.key)
+        taker_action.sign(self.signer.key)
+
+        # Create request parameters
+        maker_params = {
+            "direction": maker_action.module_data.get_direction(),
+            "instrument_name": instrument_name,
+            **maker_action.to_json(),
+        }
+
+        taker_params = {
+            "direction": taker_action.module_data.get_direction(),
+            "instrument_name": instrument_name,
+            **taker_action.to_json(),
+        }
+
+        payload = {
+            "wallet": self.wallet,
+            "maker_params": maker_params,
+            "taker_params": taker_params,
+        }
+
+        response_data = self._send_request(url, json=payload)
+
+        # Extract transaction_id from response for polling
+        if "result" in response_data and "transaction_id" in response_data["result"]:
+            transaction_id = response_data["result"]["transaction_id"]
+            return wait_until(
+                self.get_transaction,
+                condition=_is_final_tx,
+                transaction_id=transaction_id,
+            )
```

Comment:
Id say either;

i) make this parse block and wait a util function, as you re-use it elsewhere.;